### PR TITLE
Some tweaks to the mfem::Device class [okina-device-tweaks]

### PR DESCRIPTION
**Pull request by v-dobrev**

### CODE REVIEW

File: examples/ex1.cpp [hunk 1]
```diff
@@ -25,12 +25,13 @@
...
-//               ex1 -p -d 'omp raja'
-//               ex1 -p -d 'omp occa'
-//               ex1 -m ../data/beam-hex.mesh -p -d cuda
+// Device sample runs:
+//             > ex1 -p -d cuda
+//             > ex1 -p -d raja-cuda
+//             > ex1 -p -d occa-cuda
+//             > ex1 -p -d raja-omp
+//             > ex1 -p -d occa-omp
+//             > ex1 -m ../data/beam-hex.mesh -p -d cuda
```
Comment (tzanio) [inline examples/ex1.cpp]: We will have to start testing these too at some point...

Comment (tzanio) [reply examples/ex1.cpp]: With `>` the testing scripts ignore the sample run

**Comment by v-dobrev** [comment]
@camierjs, I added the `>` in front of the device sample runs to prevent the script `config/sample-runs.sh` from detecting and trying to run them. The pattern used there in described here: https://github.com/mfem/mfem/blob/5b269399ed838cf82c5b4d683df7ec0b551699c6/config/sample-runs.sh#L137-L138 and here: https://github.com/mfem/mfem/blob/5b269399ed838cf82c5b4d683df7ec0b551699c6/config/sample-runs.sh#L142-L143

File: general/device.hpp [hunk 2]
```diff
@@ -17,114 +17,174 @@
...
+      RAJA_CUDA = 1 << 5,
+      /** @brief [host] OCCA CPU backend: sequential execution on each MPI rank.
+          Enabled when MFEM_USE_OCCA = YES. */
+      OCCA_CPU = 1 << 6,
+      /// [host] OCCA OpenMP backend. Enabled when MFEM_USE_OCCA = YES.
+      OCCA_OMP = 1 << 7,
+      /** @brief [device] OCCA CUDA backend. Enabled when MFEM_USE_OCCA = YES
+          and MFEM_USE_CUDA = YES. */
+      OCCA_CUDA = 1 << 8
+   };
```
Comment (tzanio) [inline general/device.hpp]: Very nice documentation

File: general/mem_manager.hpp [hunk 3]
```diff
@@ -19,7 +19,7 @@
 
 #include "globals.hpp"
 
-using std::size_t;
+using std::size_t; // FIXME: this should not be here
```
Comment (tzanio) [inline general/mem_manager.hpp]: Can we remove it then?

File: general/device.hpp [hunk 4]
```diff
@@ -17,114 +17,174 @@
...
+   /// Return true if the Device is enabled.
+   static inline bool IsEnabled() { return Get().mode == DEVICE; }
 
-   static inline bool UsingDevice() { return DeviceEnabled() && Get().mode == DEVICE; }
-   static inline bool UsingHost() { return !UsingDevice(); }
+   /// The opposite of IsEnabled().
+   static inline bool IsDisabled() { return !IsEnabled(); }
 
+   // FIXME: what is "tracking"?
    static inline void DisableTracking() { Get().isTracking = false; };
```
Comment (tzanio) [inline general/device.hpp]: @camierjs can you comment on this? Doesn't look like it is really used...

File: general/mem_manager.hpp (hunk 3 above)
Comment (v-dobrev) [reply general/mem_manager.hpp]: I'll fix it.

**Comment by dmed256** [comment]
🎉  Looks pretty good!

File: examples/ex1p.cpp [hunk 5]
```diff
@@ -209,11 +210,6 @@ int main(int argc, char *argv[])
 
    a->FormLinearSystem(ess_tdof_list, x, *b, A, X, B);
 
-   if (myid == 0)
-   {
-      cout << "Size of linear system: " << A->Height() << endl;
-   }
-
```
Comment (tzanio) [inline examples/ex1p.cpp]: This is actually a "feature" -- currently the `Operator` sizes are the local ones. We can decide to make them global, but that should be a separate change. Given that the output is the same as "Number of finite element unknowns", I think it is simplest to just take this out.

**Pull request merged**